logfire/_internal/stack_info.py

```diff
@@ -65,3 +73,35 @@ def get_caller_stack_info(stack_offset: int = 3) -> StackInfo:
         return get_stack_info_from_frame(frame)
     except Exception:  # pragma: no cover
         return {}
+
+
+def get_user_stack_offset() -> int:
+    """Get the stack offset of the user code.
+
+    We want to skip the internal code, and third party code, and get the user code stack info.
+
+    Returns:
+        The stack offset of the user code.
+    """
+    stack_offset = 2
+    try:
+        frame = inspect.currentframe()
+        while frame and not is_user_filename(frame.f_code.co_filename):
+            frame = frame.f_back
+            stack_offset += 1
+    except Exception:  # pragma: no cover
+        pass
+    return stack_offset
+
+
+@lru_cache(maxsize=2048)
+def is_user_filename(filename: str) -> bool:
+    """Check if the filename is a user filename.
+
+    Args:
+        filename: The filename to check.
+
+    Returns:
+        True if the filename is a user filename, False otherwise.
+    """
+    return not filename.startswith(PREFIXES)
```

Comment:
```suggestion
    return not Path(filename).absolute().startswith(PREFIXES)
```

I'm not sure if `frame.f_code.co_filename` is always absolute.

Reply:
I think it is? https://github.com/python/cpython/issues/64642

Reply:
It looks like that was released in 3.9. And since this is cached I don't think it hurts to be safe.